compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1429,7 +1431,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             span.remove_mark();
         }
         let mut expr_finder = super::FindExprBySpan::new(span);
-        let Some(hir::Node::Expr(body)) = self.tcx.hir().find(obligation.cause.body_id) else {
+        let Some(hir::Node::Expr(body)) = self.tcx.hir().find_by_def_id(obligation.cause.body_id) else {
             return false;
         };
         expr_finder.visit_expr(&body);
```

Comment:
```suggestion
        let Some(body_id) = self.tcx.hir().maybe_body_owned_by(obligation.cause.body_id) else {
            return false;
        };
        let body = self.tcx.hir().body(body_id);
        expr_finder.visit_expr(body.value);
```

Reply:
yeah, I think we need also to keep the check of `Some(hir::Node::Expr(body))` but I had this change in my local history, thanks!

Reply:
No i'm wrong, sorry I did not full think about your refactoring, you are right! thanks!